### DISCUSSION TITLE
What did you install to/from?

### DISCUSSION THREAD

**Discussion by CrappyGamingXD** [CodeRunner5235/Opencore-Steam-Deck#1] [Q&A]
Did you install this on a disk in the microSD slot to the main SSD? I'm trying to get this to work but most of the issues seem to be because I am trying to install to an SD card. I can get it installed, i just cant get first boot.

**Comment by CodeRunner5235** [comment]
Morning (or whatever it is where you are)

AFAIK, the SD card reader is connected in a way macOS doesnt support atm. You can load OpenCore and a macOS online recovery image, but macOS doesnt have drivers so an offline image or a full install will fail to boot. If you have a deck with an NVME (ie, not the 64gb version) you can dual boot on that if you know how to partition it, or you can install on an external drive (or SD card) over the the type C port if you have an adapter or hub

**Comment by CrappyGamingXD** [comment]
I had an SD card in a type C hub which did work for install but then didn’t boot. It usually resulted in “uiminitialize unable to create filterinterrupteventsource”. One time I did get as far as a crossed out circle icon on screen with scrambled text. Do you recognize either of these errors or did you run into these at all?

**Reply by CodeRunner5235** [reply]
Sounds like a USB issue. There does seem to be a racing condition, so if it freezes early in the boot process, a reboot should fix. But check your boot args for vsmcgen=1.
the scrambled text and prohibitory sign is most likely due to macOS being unable to find the root device. If your booting from the USB-C port, double check your maps (USBtoolbox kexts and the gux-ryzenusbfix)

**Reply by CrappyGamingXD** [reply]
I have vsmcgen=1 in my boot args, and I do have the USBtoolbox and gux-ryzenxhcifix kexts already, and theyre pointed to in my config.plist. Would there be anything else I did wrong for those to not be working?

**Reply by CodeRunner5235** [reply]
How is USB configured in the BIOS? Is it DRD or XHCI?

**Reply by CrappyGamingXD** [reply]
My USB is in XHCI mode. After reading this I tried it in DRD mode and got the error "build_userspace_exit_reason: illegal flags passed from userspace (some masked off) 0x01, ns:20, code 0x1". This is another error I am getting fairly often.

**Reply by CodeRunner5235** [reply]
Can you send your config.plist as well as a picture of the screen when it hangs or panics? Take out any smbios data

**Reply by CrappyGamingXD** [reply]
[config.plist.txt](https://github.com/user-attachments/files/17298212/config.plist.txt)

![PXL_20241008_175958952](https://github.com/user-attachments/assets/fecd0ec0-bf29-4e97-a69a-d2ff628fedf5)

**Comment by CodeRunner5235** [comment]
looks like you have alcid=1 and -no_compat_check in your boot args. Not sure what -no_compat_check is for but I know alcid=1 is for appleALC.kext, which you dont have. try removing those two from your boot arguments

**Reply by CrappyGamingXD** [reply]
Doesn't seem to have changed anything. Any other things I should try?

**Comment by CodeRunner5235** [comment]
Try with just -v keepsyms=1 and vsmcgen=1

**Reply by CrappyGamingXD** [reply]
Still no change. Any chance you can send me your EFI folder and I can add my own smbios stuff to see if its my setup/bios settings that are wrong?

**Comment by CodeRunner5235** [comment]
I can sent you one of my known working plists for you to test and compare against
[config.plist.txt](https://github.com/user-attachments/files/17298953/config.plist.txt)

**Reply by CrappyGamingXD** [reply]
I tested that with my smbios information and got stuck on DSMOS has arrived, edited it to include the AppleMCEReporterDisabler.kext then got the same build userspace exit reason error.

**Reply by CodeRunner5235** [reply]
What macOS version?

**Reply by CrappyGamingXD** [reply]
I originally tried Big Sur because that's what I saw you used, saw online that Big Sur doesn't like being on an SD card then moved to High Sierra. Should I go back to trying Big Sur?

**Comment by CodeRunner5235** [comment]
Yes, High Sierra is your issue. Try a version like catalina or newer

**Reply by CrappyGamingXD** [reply]
Big Sur installed but wont boot, stuck on Waiting for DSMOS...
This is another common issue I struggled with before changing to High Sierra
![PXL_20241009_041909423](https://github.com/user-attachments/assets/58dcd61e-5f43-4093-9b2d-162b04a83c13)

**Reply by CodeRunner5235** [reply]
There are some points in the recovery boot process that take a hot minute, this may be one of them. I dont know how long your giving it, but try waiting a few minutes to see if it changes. It takes around 5 one you get into the unsupported PCH area. A macOS install is much quicker on the boot (depending on storage and such). 

You can also try removing the reporterdisabler. It isnt needed until Monterey and up anyway

**Reply by CrappyGamingXD** [reply]
I removed reporterdisabler already, and I let it sit for about 2 hours on this screen, no changes.

**Reply by CodeRunner5235** [reply]
Stalling at dsmos (dont steal macOS) points to something being up with your virtualSMC. You already have the boot arg, so make sure lilu is before vitualSMC in the plist (it should be) and everything is up to date. If it gives issues still, send your latest OC boot log

**Reply by CrappyGamingXD** [reply]
I checked and my lilu is before my virtualSMC. I also have tried replacing my virtualSMC already so heres my latest boot log.
  
[opencore-2024-10-10-002220.txt](https://github.com/user-attachments/files/17320190/opencore-2024-10-10-002220.txt)

**Reply by CodeRunner5235** [reply]
Strange. Maybe it is because you are booting from and SD card. Can you try a USB hard drive? If not, can you try installing catalina? Everyone else I know who got it working was on a USB hard drive or on the internal NVMe, so you may be on to something with big sur not liking SD cards. Catalina tends to be much more forgiving in my experience

**Reply by CrappyGamingXD** [reply]
Tried Catalina and made negative progress. Now I can't even get into the installer, stuck on Unable to create filterInterruptEventSource. I remade the installer SD, once again using your config.plist and updating all the kexts and drivers to the latest versions. Here is the opencore log. Is there something I messed up on remaking the installer?

[opencore-2024-10-10-192219.txt](https://github.com/user-attachments/files/17333294/opencore-2024-10-10-192219.txt)

**Reply by CrappyGamingXD** [reply]
Just tried putting the installer onto a flash drive as well and no luck there either. 
![PXL_20241010_215347369 MP](https://github.com/user-attachments/assets/b10c63e0-bcb4-495a-955e-5302388c80a5)
This is still the steam deck just on an external display

**Reply by CodeRunner5235** [reply]
Interesting. Can you sent me your current config.plist again and ill test on my hardware? Want to narrow this down if its in opencore or something specific to your setup. Also, you mind finding your bios version while your at it?

**Reply by CrappyGamingXD** [reply]
Steam deck bios version F7A0120 (LCD 512gb if that matters)

[config.plist.txt](https://github.com/user-attachments/files/17334996/config.plist.txt)

**Reply by CrappyGamingXD** [reply]
I just tried making a new UTBMap kext using the windows tool and am still getting the same errors. Idk if this helps or not, just more information to work with

**Reply by CrappyGamingXD** [reply]
By using an old version of my conflig.plist ive gotten big sur installed to a usb drive but am once again stuck at Waiting for DSMOS... Is there anything I need to do besides having VirtualSMC.kext, having it in my conflig.plist and having vsmcgen=1 in my boot args? Has there been any success previously using FakeSMC?

**Reply by CodeRunner5235** [reply]
On vacation right now, so I cant do much testing. That should be all you need to get virtualSMC working. I dont see the harm in trying fakeSMC. I haven't tested it myself

**Reply by CodeRunner5235** [reply]
[EFI.zip](https://github.com/user-attachments/files/17368974/EFI.zip)
Actually, give this EFI a shot. If it works, can you send me yours so I can poke at it?

**Reply by CrappyGamingXD** [reply]
That EFI did not work unfortunately, same behavior as usual. I can boot the installer and fully install, but once I try the first boot i get the UIMinitialize unable to create filterinterrupteventsource error. Are there any BIOS settings I forgot to change? Could it be my USBC hub?

**Reply by CodeRunner5235** [reply]
So we know it isnt an opencore configuration issue. I would look at your bios settings and version, try updating it. If that doesnt work, it may be your hub or something else is the issue

**Comment by lucazsh** [comment]
Hey @CodeRunner5235! I need help formatting my SSD on macOS recovery it can't format it correctly (the pc freezes and restarts), I don't know what it is, and when I succeed and install mac os on it, I get to the Apple logo and about 29 minutes and then after some time it reboots and starts again. Thanks!
